--- /boot/grub/grub.cfg.bak	2024-03-23 16:30:52.476335784 +0100
+++ /boot/grub/grub.cfg	2024-03-23 16:30:52.519669089 +0100
@@ -89,7 +89,7 @@
 ### END /etc/grub.d/00_header ###
 
 ### BEGIN /etc/grub.d/10_linux ###
-menuentry 'aerOS Linux' --class aeros --class gnu-linux --class gnu --class os $menuentry_id_option 'gnulinux-simple-04e5693f-496c-418a-b675-f908d2c4331c' {
+menuentry 'aerOS' --class aeros --class gnu-linux --class gnu --class os $menuentry_id_option 'gnulinux-simple-04e5693f-496c-418a-b675-f908d2c4331c' {
 	load_video
 	set gfxpayload=keep
 	insmod gzio
@@ -106,27 +106,42 @@
 	echo	'Loading initial ramdisk ...'
 	initrd	/boot/initramfs-linux.img
 }
-submenu 'Advanced options for aerOS Linux' $menuentry_id_option 'gnulinux-advanced-04e5693f-496c-418a-b675-f908d2c4331c' {
-	menuentry 'aerOS Linux, with Linux linux' --class aeros --class gnu-linux --class gnu --class os $menuentry_id_option 'gnulinux-linux-advanced-04e5693f-496c-418a-b675-f908d2c4331c' {
-		load_video
-		set gfxpayload=keep
-		insmod gzio
-		insmod part_gpt
-		insmod ext2
-		set root='hd0,gpt2'
-		if [ x$feature_platform_search_hint = xy ]; then
-		  search --no-floppy --fs-uuid --set=root --hint-bios=hd0,gpt2 --hint-efi=hd0,gpt2 --hint-baremetal=ahci0,gpt2  04e5693f-496c-418a-b675-f908d2c4331c
-		else
-		  search --no-floppy --fs-uuid --set=root 04e5693f-496c-418a-b675-f908d2c4331c
-		fi
-		echo	'Loading Linux linux ...'
-		linux	/boot/vmlinuz-linux root=UUID=04e5693f-496c-418a-b675-f908d2c4331c rw  quiet splash loglevel=3
-		echo	'Loading initial ramdisk ...'
-		initrd	/boot/initramfs-linux.img
-	}
-}
 
+menuentry 'aerOS with safe graphics driver' --class aeros --class gnu-linux --class gnu --class os $menuentry_id_option 'gnulinux-simple-04e5693f-496c-418a-b675-f908d2c4331c' {
+	load_video
+	set gfxpayload=keep
+	insmod gzio
+	insmod part_gpt
+	insmod ext2
+	set root='hd0,gpt2'
+	if [ x$feature_platform_search_hint = xy ]; then
+	  search --no-floppy --fs-uuid --set=root --hint-bios=hd0,gpt2 --hint-efi=hd0,gpt2 --hint-baremetal=ahci0,gpt2  04e5693f-496c-418a-b675-f908d2c4331c
+	else
+	  search --no-floppy --fs-uuid --set=root 04e5693f-496c-418a-b675-f908d2c4331c
+	fi
+	echo	'Loading Linux linux ...'
+	linux	/boot/vmlinuz-linux root=UUID=04e5693f-496c-418a-b675-f908d2c4331c rw  quiet splash nomodeset loglevel=3
+	echo	'Loading initial ramdisk ...'
+	initrd	/boot/initramfs-linux.img
+}
 ### END /etc/grub.d/10_linux ###
+menuentry 'aerOS in TTY mode' --class aeros --class gnu-linux --class gnu --class os $menuentry_id_option 'gnulinux-simple-04e5693f-496c-418a-b675-f908d2c4331c' {
+	load_video
+	set gfxpayload=keep
+	insmod gzio
+	insmod part_gpt
+	insmod ext2
+	set root='hd0,gpt2'
+	if [ x$feature_platform_search_hint = xy ]; then
+	  search --no-floppy --fs-uuid --set=root --hint-bios=hd0,gpt2 --hint-efi=hd0,gpt2 --hint-baremetal=ahci0,gpt2  04e5693f-496c-418a-b675-f908d2c4331c
+	else
+	  search --no-floppy --fs-uuid --set=root 04e5693f-496c-418a-b675-f908d2c4331c
+	fi
+	echo	'Loading Linux linux ...'
+	linux	/boot/vmlinuz-linux root=UUID=04e5693f-496c-418a-b675-f908d2c4331c rw  loglevel=3 $vt_handoff 3
+	echo	'Loading initial ramdisk ...'
+	initrd	/boot/initramfs-linux.img
+}
 
 ### BEGIN /etc/grub.d/15_ostree ###
 ### END /etc/grub.d/15_ostree ###
